editor/export/editor_export_platform.cpp

```diff
@@ -230,17 +230,23 @@ Error EditorExportPlatform::_save_pack_file(void *p_userdata, const String &p_pa
 	sd.size = p_data.size();
 	sd.encrypted = false;
 
-	for (int i = 0; i < p_enc_in_filters.size(); ++i) {
-		if (p_path.matchn(p_enc_in_filters[i]) || p_path.replace("res://", "").matchn(p_enc_in_filters[i])) {
+	if (!p_key.is_empty()) {
+		if (PackedData::file_requires_encryption(p_path)) {
 			sd.encrypted = true;
-			break;
-		}
-	}
+		} else {
+			for (int i = 0; i < p_enc_in_filters.size(); ++i) {
+				if (p_path.matchn(p_enc_in_filters[i]) || p_path.replace("res://", "").matchn(p_enc_in_filters[i])) {
+					sd.encrypted = true;
+					break;
+				}
+			}
 
-	for (int i = 0; i < p_enc_ex_filters.size(); ++i) {
-		if (p_path.matchn(p_enc_ex_filters[i]) || p_path.replace("res://", "").matchn(p_enc_ex_filters[i])) {
-			sd.encrypted = false;
-			break;
+			for (int i = 0; i < p_enc_ex_filters.size(); ++i) {
+				if (p_path.matchn(p_enc_ex_filters[i]) || p_path.replace("res://", "").matchn(p_enc_ex_filters[i])) {
+					sd.encrypted = false;
+					break;
+				}
+			}
```

Comment:
This can use `file_require_encryption(...)` in `encryption_export_util` to avoid duplication.